snuba/datasets/entities/sessions.py

```diff
@@ -250,3 +257,59 @@ def get_query_processors(self) -> Sequence[QueryProcessor]:
                 {"bucketed_started": "started"}, ("started", "received")
             ),
         ]
+
+
+GRANULARITY_MAPPING = {
+    "toStartOfMinute": 60,
+    "toStartOfHour": 3600,
+    "toDate": 86400,
+}
+
+
+def extract_granularity_from_query(query: Query) -> Optional[int]:
```

Comment:
This is useful. I would actually the part of this function that defines the patterns and applies them to an expression into `timeseries_processor.py` as they could be reused and that would make it very hard to update the processor forgetting the matchers.